src/Mod/PartDesign/Gui/Utils.cpp

```diff
@@ -153,6 +141,85 @@ PartDesign::Body *getBody(bool messageIfNot, bool autoActivate, bool assertModer
     return activeBody;
 }
 
+PartDesign::Body * makeBodyActive(App::DocumentObject* body, App::Document *doc)
+{
+    App::DocumentObject *parent = 0;
+    std::string sub;
+
+    for(auto &v : body->getParents()) {
+        if(v.first->getDocument()!=doc)
+            continue;
+        if(parent) {
+            body = 0;
+            break;
+        }
+        parent = v.first;
+        sub = v.second;
+    }
+
+    if(body) {
+        auto _doc = parent?parent->getDocument():body->getDocument();
+        _FCMD_DOC_CMD(Gui, _doc, "ActiveView.setActiveObject('" << PDBODYKEY
+                      << "'," << Gui::Command::getObjectCmd(parent?parent:body)
+                      << ",'" << sub << "')");
+        // return Gui::Application::Instance->activeView()->
+        //     getActiveObject<PartDesign::Body*>(PDBODYKEY,topParent,subname);
+    }
+
+    return dynamic_cast<PartDesign::Body*>(body);
+}
+
+PartDesign::Body * needActiveBodyMessage (App::Document *doc,
+                                          const QString& infoText)
+{
+    PartDesign::Body* activeBody = nullptr;
+    QDialog dia(Gui::getMainWindow());
+    PartDesignGui::Ui_DlgActiveBody dlg;
+    dlg.setupUi(&dia);
+
+    QObject::connect(dlg.bodySelect, SIGNAL(itemDoubleClicked(QListWidgetItem *)),
+                     &dia, SLOT(accept()));
+
+    if(!infoText.isEmpty()) {
+        dlg.label->setText(infoText + QObject::tr("\n\n") +
+                           QObject::tr("Please select"));
+    }
+
+    auto bodies = doc->getObjectsOfType(PartDesign::Body::getClassTypeId());
+
+    PartDesign::Body* bodyOfActiveObject = nullptr;
+    for (Gui::SelectionSingleton::SelObj obj :  Gui::Selection().getSelection()) {
+        bodyOfActiveObject = PartDesign::Body::findBodyOf(obj.pObject);
+        break; // Just get the body for first selected object
+    }
+
+    for (auto body : bodies) {
+        auto item = new QListWidgetItem(QString::fromUtf8(body->Label.getValue()));
+        item->setData(Qt::UserRole, QVariant::fromValue(body));
+        dlg.bodySelect->addItem(item);
+
+        if (body == bodyOfActiveObject) {
+            item->setSelected(true);
+        }
+
+        // TODO: Any other logic (hover, select effects on view etc.)
+    }
+
+    int result = dia.exec();
+    auto selectedItems = dlg.bodySelect->selectedItems();
+    if (result == QDialog::DialogCode::Accepted && !selectedItems.empty()) {
+        App::DocumentObject* selectedBody =
+            selectedItems[0]->data(Qt::UserRole).value<App::DocumentObject*>();
+        if (selectedBody) {
+            activeBody = makeBodyActive(selectedBody, doc);
+        } else {
+            return makeBody(doc);
+        }
```

Comment:
@chennes I was talking of this if-else statement. should I be removing the braces?

Reply:
According to that standard, yes, those curlies should go. But again, it's not like that's an "official FreeCAD standard" or anything. So there's no rule that says you have to make the change.